### PR TITLE
Avoid Confirmation Popup on Save Square Settings

### LINKED ISSUE #208: Confirmation Popup appears when Saving Square Settings
### Describe the bug

Related internal discussion is [here](https://fueled10up.slack.com/archives/C01UCQGPBDJ/p1724833450908149).

Each time the save button is clicked, a confirmation popup appears before the page reloads. This occurs because the page is being [reloaded](https://github.com/woocommerce/woocommerce-square/pull/105/commits/9d878b42afa5cb6d800e5b29d15049a291bde1ef) once the settings are saved to populate new locations in the dropdown (if environment is changed). This behavior can be improved by using client-side rendering to update the dropdown without requiring a page reload.

### To reproduce

1. Navigate to WooCommerce > Settings > Square.
2. Change the environment to Sandbox and save.
3. The page reloads.
4. Change the environment to Production (no need to connect) and save.
5. Notice the confirmation popup.

### Expected behavior

The Square settings should save without triggering a confirmation popup, and no page reload should be required.

### Screenshots

confirmation-popup

**Environment (please complete the following information):**
 - WordPress Version
 - WooCommerce Version
 - WooCommerce Square Plugin Version
 - Browser [e.g. chrome, safari] and Version
 - Any other plugins installed

### Additional details

<details><summary>System status</summary>
  
  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```
</details>

### PR DESCRIPTION
**Pull request by faisal-alvi**
### All Submissions:

* [ ] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

On the Square settings page, when the merchant changes the environment from the dropdown, it’s necessary to refresh the list of available locations based on the selected environment. Currently, this is achieved by reloading the page after saving, allowing the merchant to see and select the new location. However, based on an [internal discussion](https://fueled10up.slack.com/archives/C01UCQGPBDJ/p1724833450908149) about fixing the confirmation popup issue, it was decided to add a "Connect to Square Sandbox" button. This button will connect to the sandbox environment and populate the new locations, allowing users to select sandbox locations before clicking the main Sae button, thereby avoiding the need for a reload.

The original issue was that a confirmation popup appeared when the "Save Settings" button was selected, caused by the page reload request (see https://github.com/woocommerce/woocommerce-square/pull/105/commits/9d878b42afa5cb6d800e5b29d15049a291bde1ef). The core behavior, which uses the native save button, does not trigger this confirmation popup.

After further investigation, it was observed that even with the addition of the "Connect to Square Sandbox" button and the removal of the reload request on save, the confirmation popup still appears when attempting to navigate away from the page. This made the new button less effective.

This PR resolves the issue by triggering a click on the core save button (which is hidden) when the React-based save button is clicked, instead of reloading the page. This approach ultimately prevents the confirmation popup from appearing.

Closes #208.

### Steps to test the changes in this Pull Request:

1. Navigate to WooCommerce > Settings > Square.
2. Change the environment to Sandbox and save.
3. The page reloads.
4. Change the environment to Production (no need to connect) and save.
5. Notice the page saves and no confirmation popup appears.

### Changelog entry

> Fix - Confirmation popup appears when saving the Square settings.

### CODE REVIEW

**Comment by faisal-alvi** [comment]
The failing checks appear to be due to unrelated reasons.

**Review by dkotter — APPROVED** [review]

**Comment by qasumitbagthariya** [comment]
### QA Update ✅ 

-------------

I have verified this PR in the `fix/208` branch which has been fixed and is functioning as intended.

**Testing Environment** 

<Details>

- WordPress: 6.6.1
- Theme: Storefront 4.6.0
- Theme: Twenty Twenty-Four 1.2
- WooCommerce - 9.2.3
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Branch: fix/208

</Details>


https://github.com/user-attachments/assets/64ed5249-02b2-49ff-859c-e060b978adf9



Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for code review (Woo)
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [ ] Yes
- [x] Not Required/Applicable for this PR
------------------------------------------------

**Review by qasumitbagthariya — APPROVED** [review]

**Review by wjrosa — APPROVED** [review]
The code changes look good to me!

File: package-lock.json
Comment (jeffpaul) [inline package-lock.json:1]: Were the changes here intentional?

Comment (dkotter) [reply package-lock.json:1]: Yes. The `package-lock.json` file was outdated and was causing our automated tests to fail. So not needed as part of this PR but is needed to get those tests to run

**Comment by qasumitbagthariya** [comment]
Regression / Smoke Test Report  ✅ 
-------

Tested with Archive File created via "php woorelease.phar build repo_URL" (Composer version 2.5.5, npm version 8.19.4, node version 16.20.0)

Status- Working expected with Plugin Archive/Zip file same as fix specific branch.


Next Step- Ready to Merge 🚀

**Pull request merged**